src/Dfc.CourseDirectory.Functions/GetAllPublishedFrameworkApprenticeshipIds.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Dfc.CourseDirectory.Functions
+{
+    public class GetAllPublishedFrameworkApprenticeshipIds
+    {
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+
+        public GetAllPublishedFrameworkApprenticeshipIds(ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher ?? throw new ArgumentNullException(nameof(cosmosDbQueryDispatcher));
+        }
+
+        [FunctionName("GetAllPublishedFrameworkApprenticeshipIds")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest req, ILogger log)
+        {
+            var results = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeships(a => a.FrameworkId != null));
```

Comment:
Need an extra condition in the predicate here to check status (or the query is mis-named); `GetApprenticeships` doesn't look to be only returning published records

Reply:
https://github.com/SkillsFundingAgency/dfc-coursedirectory/pull/1717/files#diff-4ecd5d0b7c870949dc17394bc2c3e3efad0b75be40cf3fe318098835a0ceef51R21?

Reply:
That's not the same as 'Published'

Reply:
Given the scope and intention of the story, I think we want to archive all Frameworks that aren't either `Archived` or `Deleted`. So I think the problem here is with the naming of the function, not the execution. What would you refer to for all Apprenticeships that are neither `Archived` or `Deleted`?

Reply:
We don't really have a word for that. That said, we always exclude `Archived` and `Deleted` so for me that part is implied. I'd go with something like `GetApprenticeshipIdsWithFramework`.

Reply:
I've renamed it to `GetAllFrameworkApprenticeshipIds` - Frameworks will be basically dead once this story is completed, as such so will this function.